client/gutenberg/editor/without-iframe.js

```diff
@@ -0,0 +1,135 @@
+
+import '@wordpress/components/build-style/style.css';
+import '@wordpress/block-editor/build-style/style.css';
+import '@wordpress/block-library/build-style/style.css';
+import '@wordpress/block-library/build-style/editor.css';
+import '@wordpress/block-library/build-style/theme.css';
+import '@wordpress/format-library/build-style/style.css';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import EditorDocumentHead from 'post-editor/editor-document-head';
+import { Placeholder } from './placeholder';
+import './without-iframe.scss';
+import PerformanceTrackerStop from 'lib/performance-tracking/performance-tracker-stop';
+import Layout from '@wordpress/edit-post/src/components/layout';
+
+const getStatsPath = ( { postId } ) =>
+	postId
+		? '/block-editor/without-iframe/:post_type/:site/:post_id'
+		: '/block-editor/without-iframe/:post_type/:site';
+
+const getStatsTitle = ( { postId, postType } ) => {
+	let postTypeText;
+
+	switch ( postType ) {
+		case 'post':
+			postTypeText = 'Post';
+			break;
+		case 'page':
+			postTypeText = 'Page';
+			break;
+		default:
+			postTypeText = 'Custom Post Type';
+			break;
+	}
+
+	return postId
+		? `Block Editor > ${ postTypeText } > Edit`
+		: `Block Editor > ${ postTypeText } > New`;
+};
+
+const getStatsProps = ( { postId, postType } ) =>
+	postId ? { post_type: postType, post_id: postId } : { post_type: postType };
+
+function Gutenberg( props ) {
+	const { postId, siteId, postType } = props;
+	const [ post, setPost ] = useState( null );
+	const [ blocks, updateBlocks ] = useState( null );
+
+	const { toggleFeature } = useDispatch( 'core/edit-post' );
+	const { setupEditor } = useDispatch( 'core/editor' );
+
+	useEffect( () => {
+		registerCoreBlocks();
+		toggleFeature( 'fullscreenMode' );
+	}, [ toggleFeature ] );
+
+	useEffect( () => {
+		// retrieve the full post description including "raw" content to access block markup
+		wpcom.site( siteId ).post( postId ).get( { context: 'edit' } ).then( setPost );
+	}, [ postId, siteId ] );
+
+	useEffect( () => {
+		if ( ! post ) {
+			return;
+		}
+
+		setupEditor( post, {} );
+
+		const parsedBlocks = parse( post.content );
+		// give the blocks and initial fake clientId of their starting index.
+		updateBlocks( parsedBlocks );
+	}, [ post, setupEditor ] );
+
+	if ( ! post || ! blocks ) {
+		return <Placeholder />;
+	}
+
+	return (
+		<>
+			<PageViewTracker
+				path={ getStatsPath( props ) }
+				title={ getStatsTitle( props ) }
+				properties={ getStatsProps( props ) }
+			/>
+			<EditorDocumentHead />
+			<PerformanceTrackerStop />
+
+			<div className="editor__without-iframe" role="main">
+				<h1>Gutenberg In Calypso</h1>
+
+				<SlotFillProvider>
+					<DropZoneProvider>
+						<BlockContextProvider value={ { postId, postType } }>
+							<BlockEditorProvider
+								value={ blocks }
+								onInput={ updateBlocks }
+								onChange={ updateBlocks }
```

Comment:
It seems that the inserter crash is happening because the block editor is expecting some settings to exist, but they don't exist in Calypso. I don't know if it makes sense having those settings in Calypso, but an easy solution is just passing them to the `BlockEditorProvider` component:

```suggestion
								onChange={ updateBlocks }
								settings={ {
									__experimentalBlockPatternCategories: [],
									__experimentalBlockPatterns: [],
								} }
```

But it looks like a bug on Gutenberg. It shouldn't be expecting those settings to be set, especially since they're experimental settings. I'll propose a change there.

Reply:
Thanks @diegohaz! That fixed it perfectly!